src/colmap/geometry/pose.cc

```diff
@@ -153,6 +154,27 @@ Rigid3d InterpolateCameraPoses(const Rigid3d& cam_from_world1,
                  cam_from_world1.translation + translation12 * t);
 }
 
+Eigen::Quaterniond QuaternionFromAngleAxis(const Eigen::Vector3d& omega) {
+  double half_angle = 0.5 * omega.norm();
+  Eigen::Vector3d axis = omega.normalized();
+  Eigen::Quaterniond q(Eigen::AngleAxisd(half_angle, axis));
+  return q;
+}
```

Comment:
1. We unnecessarily compute the norm of omega twice.
2. omega.normalized() can lead to NaNs.

Reply:
Thanks. This is updated.

Reply:
This was actually a bug. AngleAxisd should receive the full angle rather than half angle. And this is fixed now.